README.md

```diff
@@ -0,0 +1,163 @@
+# Aligent Shopify Theme Development Template
+
+**Version**: (beta) 1.0  
+
+## Description
+
+Docker image for running the shopify-cli command without requiring it to be installed locally. It includes setup script, few custom shopify commands that simplifies working with Shopify themes via the shopify_wrapper, and yml file for integrating with Bitbucket Pipelines. 
+
+## Limitations
+
+Currently, this setup only supports Bitbucket Pipelines. If the client's repository is hosted on platforms like GitHub, GitLab, or others, the deployment pipeline functionality will not be available and would need to be configured separately.
+
+## File Structure
+```bash
+.
+├── assets
+│   ├── entrypoint.sh
+│   └── shopify_wrapper.sh
+├── bitbucket-pipelines.yml
+├── docker-compose.yml
+├── Dockerfile
+├── README.md
+└── set-up.sh
+```
+
+## File Descriptions
+
+- assets/
+
+    - **entrypoint.sh**: This script is the entrypoint for the Docker container. It initializes the environment, sets the correct permissions, and pulls the Shopify theme using the specified credentials and configurations. It also handles logging and error management during startup.
+    - **shopify_wrapper.sh**: A wrapper script for the Shopify CLI that overrides certain commands to enforce specific behavior, such as preventing accidental theme pushes and appending required parameters to shopify dev up commands.
+- **bitbucket-pipeline.yml**: Template bitbucket yml
+- **docker-compose.yml**: Defines the Docker Compose configuration for the project, specifying services, environment variables, volumes, and other settings needed to run the Shopify CLI within a containerized environment. It facilitates easy orchestration and management of the container setup.
+
+- **Dockerfile**: Contains instructions to build the Docker image for the project. It sets up the environment by installing necessary dependencies, copying scripts, configuring permissions, and setting the appropriate user context for running Shopify CLI commands.
+
+- **README.md**: Provides an overview of the project, including setup instructions, usage guidelines, and other relevant documentation to help users understand and work with the project effectively.
+
+- **set-up.sh**: A setup script that initializes environment variables, prompts the user for required configuration values (like Shopify store URL, theme ID, and access token), and sets up the local environment for running Docker Compose commands. It ensures that the necessary variables are exported and available for the Docker environment and adding local development files to .gitignore.
+
+
+
+## Instructions
+
+### Step 1: Create NEW folder for your shopify development.
+
+Create a new workspace for your shopify development.
+
+### Step 2: Download this repo.
+
+Navigate to the repository and run the following command so it will not include the repository meta-data and README.md.
+
+```bash
```

Comment:
I thought we were going to publish this to docker hub? :thinking: Then the devs can just use a bash alias to execute

Reply:
The base image has been published to Docker hub. But as for the local dev env of the devs, they need a little bit more of customisation for the wrappers and file permissions.

Reply:
What's stopping us from just running the docker container with an alias without cloning everything down? The extra customisation can be built into the container